go/cli/mcap/cmd/sort.go

```diff
@@ -124,8 +124,9 @@ func sortFile(w io.Writer, r io.ReadSeeker) error {
 	}
 	schemas := make(map[uint16]*mcap.Schema)
 	channels := make(map[uint16]*mcap.Schema)
+	message := mcap.Message{}
 	for {
-		schema, channel, message, err := it.Next(nil)
+		schema, channel, _, err := it.Next2(&message)
```

Comment:
this is personal preference obviously, but I wonder if this is the calling style we should encourage vs

```
schema, channel, message, err := it.Next2(&message)
```

the effect is the same, the main difference is you can switch from the allocating to the reusing form just by changing the arg, rather than both the arg and the return value.

Reply:
I guess you'd need to change the example to declare message as a pointer value.

One other nice thing about this seems like it allows the user to handle channel, schema, and message all the same in their code, versus dealing with a pointer for channel/schema and a struct for message.